pkg/services/provisioning/models.go

```diff
@@ -0,0 +1,26 @@
+package provisioning
+
+import (
+	"errors"
+
+	"github.com/grafana/grafana/pkg/registry"
+)
+
+const (
+	DatasourcesProvisionerUID   = "DatasourcesProvisioner"
+	PluginsProvisionerUID       = "PluginsProvisioner"
+	DashboardsProvisionerUID    = "DashboardsProvisioner"
+	NotificationsProvisionerUID = "NotificationsProvisioner"
+)
+
+var (
+	ErrUnknownProvisioner = errors.New("unknown provisioner provided")
+)
+
+type ProvisioningService interface {
```

Comment:
We modified the interface to hide the Provisioners, in order to make this interface less subject to changes

Reply:
I have no super strong opinion on this, so I'd rather prefer if someone else more familiar with this code can bring some additional feedback.

However, I am a bit concerned about the implications of this new approach, because I think it makes the code less safe. I could see the benefits if this interface would be part of a public API (e.g. a library shared with world), but since it's mostly for internal usage (no actual breaking changes), I'd prefer to make it more explicit (therefore safer) even if that implies changing the interface more often. But that's just my five cents 🙈

Reply:
Thanks Joan for your input. I strongly agree with you, I see the benefits of making the interface display all provisioning methods:
* Less subject to runtime errors (provisionerUID not existing)
* More explicit: All implementations would have to provide the same provisioners for sure
However I also see:
* How annoying it might become to have to add new methods to the enterprise implementation each time the OSS one moves. (People commiting only in OSS would not even see that it broke enterprise compilation)
* That to use the enterprise provisioners we would have either to type assert the service to its enterprise implementation or to also modify the OSS interface with Enterprise provisioners.

Quite frankly, I have no strong opinion as well and I fully rely on your expertise. For maintenance I assume that the more explicit the interface, the less bugs prone.
My only focus was to avoid too many changes in the interface for maintaining both implementations, but it might not be the most important problem...

Any other opinion ?